yaml_edit/lib/src/map_mutations.dart

```diff
@@ -47,19 +44,16 @@ SourceEdit updateInMap(
 
 /// Performs the string operation on [yaml] to achieve the effect of removing
 /// the element at [key] when re-parsed.
-SourceEdit removeInMap(YamlEditor yamlEdit, YamlMap map, Object key) {
-  ArgumentError.checkNotNull(yamlEdit, 'yamlEdit');
-  ArgumentError.checkNotNull(map, 'map');
-
+SourceEdit? removeInMap(YamlEditor yamlEdit, YamlMap map, Object? key) {
   if (!containsKey(map, key)) return null;
 
   final keyNode = getKeyNode(map, key);
   final valueNode = map.nodes[keyNode];
```

Comment:
```suggestion
  final valueNode = map.nodes[keyNode]!;
```

This way it's closer to the code that ensures this will never fail..